src/pperm.c

```diff
@@ -1699,125 +1694,71 @@ Obj FuncNaturalLeqPartialPerm(Obj self, Obj f, Obj g)
                 }
             }
         }
-        else {
-            ErrorQuit("usage: the arguments must be partial perms,", 0L, 0L);
-        }
-    }
-    else {
-        ErrorQuit("usage: the arguments must be partial perms,", 0L, 0L);
     }
     return True;
 }
 
-// could add use of rank to improve things here. JDM
 Obj FuncJOIN_IDEM_PPERMS(Obj self, Obj f, Obj g)
 {
     GAP_ASSERT(IS_PPERM(f));
     GAP_ASSERT(IS_PPERM(g));
 
-    UInt   def, deg, dej, i;
-    Obj    join;
+    UInt   def, deg, i;
+    Obj    join = NULL;
     UInt2 *ptjoin2, *ptf2, *ptg2;
     UInt4 *ptjoin4, *ptf4, *ptg4;
 
-    if (EQ(f, g))
+    if (EQ(f, g)) {
         return f;
+    }
 
     def = DEG_PPERM(f);
     deg = DEG_PPERM(g);
-    dej = MAX(def, deg);
-    if (dej < 65536) {
-        join = NEW_PPERM2(dej);
-        SET_CODEG_PPERM2(join, dej);
+
+    if (def > deg) {
+        return FuncJOIN_IDEM_PPERMS(self, g, f);
```

Comment:
Nice way to simplify the following code a lot! As a minor tweak, you could also do this:
```C
  SWAP(Obj, f, g);
  SWAP(UInt, def, deg);
```

Reply:
Nice, I didn't know about `SWAP`, I've updated the PR to use this instead.